chimerapy/chimera_copy.py

```diff
@@ -0,0 +1,938 @@
+from astropy.visualization import astropy_mpl_style
+from skimage.util import img_as_ubyte
+from sunpy.coordinates import (HeliographicStonyhurst,
+                               propagate_with_solar_surface)
+from sunpy.map import Map, all_coordinates_from_map
+
+# --noverify
+
+plt.style.use(astropy_mpl_style)
+
+"""Defining the paths for example files used to run program locally"""
+
+file_path = "./"
+
+INPUT_FILES = {
+    "aia171": "http://jsoc.stanford.edu/data/aia/synoptic/2016/09/22/H1000/AIA20160922_1030_0171.fits",
+    "aia193": "http://jsoc.stanford.edu/data/aia/synoptic/2016/09/22/H1000/AIA20160922_1030_0193.fits",
+    "aia211": "http://jsoc.stanford.edu/data/aia/synoptic/2016/09/22/H1000/AIA20160922_1030_0211.fits",
+    "hmi_mag": "http://jsoc.stanford.edu/data/hmi/fits/2016/09/22/hmi.M_720s.20160922_010000_TAI.fits",
+}
+
+im171 = Map(INPUT_FILES["aia171"])
+im193 = Map(INPUT_FILES["aia193"])
+im211 = Map(INPUT_FILES["aia211"])
+imhmi = Map(INPUT_FILES["hmi_mag"])
+
+
+def rescale(proj_to: sunpy.map.Map, input_map: sunpy.map.Map):
+    """
+    Rescale the input aia image dimensions.
+
+    Parameters
+    ----------
+    proj_to: 'sunpy.map.Map'
+    input_map: 'sunpy.map.Map
+
+    Returns
+    -------
+    array: 'np.array'
+
+    """
+    with propagate_with_solar_surface():
+        map1 = proj_to.reproject_to(input_map.wcs)
+        new_x_scale = map1.scale[0].to(u.arcsec / u.pixel).value
+        new_y_scale = map1.scale[1].to(u.arcsec / u.pixel).value
+        map1.meta["cdelt1"] = new_x_scale
+        map1.meta["cdelt2"] = new_y_scale
+        map1.meta["cunit1"] = "arcsec"
+        map1.meta["cunit2"] = "arcsec"
+        return map1
+
+
+im171 = rescale(im171, im171)
+im193 = rescale(im171, im193)
+im211 = rescale(im171, im211)
+imhmi = rescale(im171, imhmi)
+
+
+def filter(map1: np.array, map2: np.array, map3: np.array):
+    """
+    Defines headers and filters aia arrays to meet header requirements. Removes 0 values from each array.
+
+    Parameters
+    ----------
+    map1: 'sunpy.map.Map'
+    map2: 'sunpy.map.Map'
+    map3: 'sunpy.map.Map'
+
+    Returns
+    -------
+    map1: 'sunpy.map.Map'
+    map2: 'sunpy.map.Map'
+    map3: 'sunpy.map.Map'
+
+    """
+    map1.data[np.where(map1.data <= 0)] = 0
+    map2.data[np.where(map2.data <= 0)] = 0
+    map3.data[np.where(map3.data <= 0)] = 0
+
+    return map1, map2, map3
+
+
+im171, im193, im211 = filter(im171, im193, im211)
+
+"""defines the shape of the arrays as "s" and "rs" as the solar radius"""
+s = np.shape(im171.data)
+# do we want the solar radius in arcsec or pixels?
+rs = im171.rsun_obs
+rs_pixels = im171.rsun_obs / im171.scale[0]
+
+
+def pix_arc(map: sunpy.map.Map):
+    """
+    Calculates the conversion value of pixel to arcsec
+
+    Parameters
+    ----------
+    map1: 'sunpy.map.Map'
+    map2: 'sunpy.map.Map'
+
+    Returns
+
+    """
+    dattoarc = map.scale[0].value
+    conver = ((s[0]) / 2) * dattoarc / map.meta["cdelt1"] - (s[1] / 2)
```

Comment:
`map.meta["cdelt1"]` is `map.scale[0]` so this `dattoarc / map.meta["cdelt1"]` should be `1` then this entire line should be 0, but the units don't make sense either
 
```
m171nrt = Map('http://jsoc.stanford.edu/data/aia/synoptic/2016/09/22/H1000/AIA20160922_1030_0171.fits')
s = m171nrt.dimensions
(s.x/2) * m171nrt.scale[0] /  m171nrt.meta["cdelt1"], (s.y / 2)
(<Quantity 512. arcsec>, <Quantity 512. pix>)
```